cmd/cluster/transferowner.go

```diff
@@ -305,17 +373,41 @@ func (o *transferOwnerOptions) run() error {
 		return fmt.Errorf("Failed to get username from new user id")
 	}
 
+	var mgmtCluster, svcCluster, hiveCluster, masterCluster *cmv1.Cluster
+
 	// Find and setup all resources that are needed
-	hiveCluster, err := utils.GetHiveCluster(o.clusterID)
+	if o.hypershift {
+		hcp, err := utils.IsHostedCluster(o.clusterID)
+		if err != nil {
+			return fmt.Errorf("cannot check if the given cluster is HCP or not")
+		}
+		if !hcp {
+			return fmt.Errorf("the given cluster is not HCP cluster")
+		}
+		mgmtCluster, svcCluster, err = utils.GetMgmtServiceCluster(o.clusterID)
+		if err != nil {
+			return err
+		}
+		masterCluster = svcCluster
```

Comment:
The masterCluster naming could be a bit confusing. Wondering if the variable can be renamed into something more fitting both for SC and Hive

Reply:
yeah, I do not think it is best name, but I am not sure what could be better. Please help suggest if you have any idea.